Client/Frontend/ContentBlocker/ContentBlockerHelper.swift

```diff
@@ -254,27 +268,31 @@ extension ContentBlockerHelper {
         }
     }
 
-    fileprivate func compileListsNotInStore(completion: @escaping () -> Void) {
+    func compileListsNotInStore(completion: @escaping (_ success: Bool) -> Void) {
         guard let ruleStore = ruleStore else { return }
         let blocklists = blocklistBasic + blocklistStrict
-        let deferreds: [Deferred<Void>] = blocklists.map { filename in
-            let result = Deferred<Void>()
```

Comment:
I think a Deferred<Maybe<Void>> makes more sense. This is also known as a "Success" inside the app. And when you want to return an error you can just pass an error to Maybe.

Reply:
A `Maybe` is clearer to an `Error?` is difficult to see why. Can you explain?
(This block is internal to the function and only concerns itself with Error -or no error)